modules/tinymce/src/core/main/ts/keyboard/NavigationUtils.ts

```diff
@@ -30,6 +31,11 @@ const moveHorizontally = (editor: Editor, direction: HDirection, range: Range, i
   const getNextPosFn = Fun.curry(CaretUtils.getVisualCaretPosition, forwards ? caretWalker.next : caretWalker.prev);
   const isBeforeFn = forwards ? isBefore : isAfter;
 
+  if (isCefAtEdgeSelected(editor)) {
+    const newRange = range.cloneRange();
+    newRange.collapse(direction === HDirection.Backwards);
+    return Optional.from(newRange);
+  }
   if (!range.collapsed) {
```

Comment:
```suggestion
  } else if (!range.collapsed) {
```
We should never have 2 if's together like this, especially when they are mutually exclusive.